src/server/data_store/lru_cache_data_store.h

```diff
@@ -0,0 +1,38 @@
+#ifndef LRU_CACHE_DATA_STORE_H
+#define LRU_CACHE_DATA_STORE_H
+
+#include <string>
+#include <unordered_map>
+#include <list>
+#include <cstdint>
+#include "data_store.h"
+
+// Immutable key value pair wrapper
```

Comment:
(just some thoughts, no need for action yet)
If you have immutable KeyValue pair.. how do we plan to implement overwriting value of the same key, and perform LRU on this?

I guess we haven't quite define what we meant by LRU, i feel like it should be LRU on keys. If the same key is overwritten, it count as a new usage..

I think for this first implementation it will be fine, but we need to define the mechanism of the LRU
https://en.wikipedia.org/wiki/Cache_replacement_policies#Least_recently_used_(LRU)